static/js/trailTraffic.js

```diff
@@ -13,84 +13,99 @@ function initializeMap() {
 $(document).ready(function() {
 
   var Segment = function(data) {
-    this.startPt = ko.observable(data.start_latlng);
-    this.endPt = ko.observable(data.end_latlng);
-    this.polyLine = ko.observable(data.map.polyline);
+    var own = this;
```

Comment:
What about calling it something like `thisSegment` rather than `own`, which is a bit generic?